src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -977,7 +977,12 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     const pkTextTracks = this._getPKTextTracks();
     const pkOffTrack = pkTextTracks.find(track => track.language === 'off');
     const getActiveVidTextTrackIndex = () => {
-      return this._nativeTextTracksMap.findIndex(textTrack => textTrack && this._getDisplayTextTrackModeString() === textTrack.mode);
+      for (const textTrackId in this._nativeTextTracksMap) {
```

Comment:
can you explain what wrong in the previous impl?

Reply:
the Array.find returns the entire object and then i need to extract the key unlike **for in** that run on the object keys and returns only what i needed from the first place, So I thought it would be more correct and readable to use it
But it is clear that both approaches will do the job